service-print/src/main/java/org/oskari/print/loader/CommandLoadImageWMTS.java

```diff
@@ -190,10 +198,51 @@ private TileMatrix getTileMatrix(TileMatrixSet tms) throws IllegalArgumentExcept
     }
 
     private TileMatrixSet getTileMatrixSet() throws IllegalArgumentException {
+        List<TileMatrixSet> possibleTileMatrixSets = new ArrayList<>();
+
         for (TileMatrixSet tms : capabilities.getTileMatrixSets()) {
-            if (srs.equals(ProjectionHelper.shortSyntaxEpsg(tms.getCrs()))) {
-                return tms;
+            String key = tms.getTileMatrixMap().keySet().iterator().next();
+            possibleTileMatrixSets.add(tms);
+        }
+
+        if (possibleTileMatrixSets.isEmpty()) {
+            throw new NullPointerException("Could not find any TileMatrixSets");
+        }
+
+        if (possibleTileMatrixSets.size() == 1) {
+            return possibleTileMatrixSets.get(0);
+        }
+
+        HashMap<String, String> alternativeTileMatrixSets = new HashMap<>();
+        JSONObject useThisTileMatrixSetInstead = layerService.find(layer.getId()).getOptions().optJSONObject("useThisAlternativeTileMatrixSet");
+
+        if (useThisTileMatrixSetInstead == null) {
+            alternativeTileMatrixSets.put(srs, srs);
+        } else if (useThisTileMatrixSetInstead.length() > 0) {
+            Iterator iterator = useThisTileMatrixSetInstead.keys();
```

Comment:
This whole else block is unnecessary. Here you're constructing a `HashMap<String, String>` out of the JSONObject which already acts as a `Map<String, String>` if you only call `JSONObject#optString(key)`.